packages/react-tabs/src/components/Tab/useTab.ts

```diff
@@ -15,24 +15,33 @@ import { SelectTabEvent } from '../TabList/TabList.types';
  * @param ref - reference to root HTMLElement of Tab
  */
 export const useTab_unstable = (props: TabProps, ref: React.Ref<HTMLElement>): TabState => {
-  const { content, icon, value } = props;
+  const { content, disabled: tabDisabled = false, icon, value } = props;
 
-  const { appearance, selected, onRegister, onUnregister, onSelect, size, vertical } = useContextSelector(
-    TabListContext,
-    ctx => ({
-      appearance: ctx.appearance,
-      selected: ctx.selectedValue === value,
-      onRegister: ctx.onRegister,
-      onUnregister: ctx.onUnregister,
-      onSelect: ctx.onSelect,
-      size: ctx.size,
-      vertical: !!ctx.vertical,
-    }),
-  );
+  const {
+    appearance,
+    listDisabled,
+    selectedValue,
+    onRegister,
+    onUnregister,
+    onSelect,
+    size,
+    vertical,
+  } = useContextSelector(TabListContext, ctx => ({
+    appearance: ctx.appearance,
+    listDisabled: ctx.disabled,
+    selectedValue: ctx.selectedValue,
+    onRegister: ctx.onRegister,
+    onUnregister: ctx.onUnregister,
+    onSelect: ctx.onSelect,
+    size: ctx.size,
+    vertical: !!ctx.vertical,
+  }));
 
-  const onClick = useEventCallback((event: SelectTabEvent) => onSelect(event, { value }));
+  const disabled = listDisabled || tabDisabled;
 
   const innerRef = React.useRef<HTMLElement>(null);
+  const mergedRef = useMergedRefs(ref, innerRef);
```

Comment:
nit: this isn't actually necessary, since you don'tuse the `mergedRef` anyhere else

```suggestion
```

Reply:
good catch!